Secret Scanning Using TruffleHog

hamzashahid57815 opened the pull request:
# 🔐 Universal TruffleHog Pre-Push Hook Installer

This PR introduces a **universal shell script** that automates the installation and management of a **TruffleHog pre-push Git hook** across platforms (Linux, macOS, and Windows Git Bash/WSL).

---

## ✅ Key Features
- **Cross-platform support** – auto-detects OS and installs TruffleHog in the correct location.  
- **Automated TruffleHog installation** – ensures developers always have the latest version available.  
- **Pre-push hook integration** – scans all staged files for secrets before allowing a push.  
- **Manual scan support** – run `./secrets-scan.sh scan <PATH>/<FILE>` for ad-hoc scans.  
- **Easy management** – includes commands to install, check status, uninstall, and scan.  

---

## 🔍 Why This Matters
- Prevents **accidental leakage of secrets** (API keys, tokens, credentials) into the repository.  
- Enforces **secure coding practices** across the team with minimal friction.  
- Provides a **consistent developer experience** by handling installation and hook setup automatically.  

---

## 📌 Usage
```bash
./secrets.sh install     # Install hook + TruffleHog
./secrets.sh status      # Check if hook is installed
./secrets.sh uninstall   # Remove hook and TruffleHog
./secrets.sh scan .      # Manual scan of current directory
```

## 📸 Screenshots

Below are some example runs of the script and TruffleHog scans:

<img width="740" height="148" alt="Screenshot 2025-08-26 113309" src="https://github.com/user-attachments/assets/b3ce56c8-9d9b-4f5e-bc2c-c466834641f7" />

drkae456 commented:
I recommend you remove emojis from running in scripts , as it's not industry standard practise

hamzashahid57815 commented:
I have removed emojis and switched to standard color-coded messages (Error/Success/Warning/Info) for cleaner, industry-standard logging. Apologies for the earlier non-standard use of emojis.
